api/groups/baseProofGroup.go

```diff
@@ -37,15 +38,40 @@ func NewProofGroup(facadeHandler data.FacadeHandler) (*proofGroup, error) {
 
 func (pg *proofGroup) getProof(c *gin.Context) {
 	rootHash := c.Param("roothash")
+	if rootHash == "" {
+		shared.RespondWith(c, http.StatusBadRequest, nil, errors.ErrEmptyRootHash.Error(), data.ReturnCodeRequestError)
+		return
+	}
+
+	address := c.Param("address")
+	if address == "" {
+		shared.RespondWith(c, http.StatusBadRequest, nil, errors.ErrEmptyAddress.Error(), data.ReturnCodeRequestError)
+		return
+	}
+
+	getProofResp, err := pg.facade.GetProof([]byte(rootHash), []byte(address))
```

Comment:
no need to cast the `address` or the `rootHash` to `[]byte` because you convert them back durring processing

Reply:
Refactored.